Add per-package landing pages and split API docs (#513)
ODINN-SciML/ODINN.jl: docs/src/Packages/odinn.md

# ODINN.jl

> [`ODINN.jl`](https://github.com/ODINN-SciML/ODINN.jl) is the top-level package of the ecosystem. It ties together the ice flow solver (`Huginn`), the mass balance models (`Muninn`), and the core data structures (`Sleipnir`) into an end-to-end differentiable glacier model. Its primary purpose is to enable **Universal Differential Equations (UDEs)**: hybrid models that combine physical PDEs with data-driven regressors such as neural networks, trained end-to-end with gradient-based optimization.

Reviewer: You are leaving behind the most important word here: inverse

Reviewer: The last part repeates what you said before, so I suggested remove it.

> [`ODINN.jl`](https://github.com/ODINN-SciML/ODINN.jl) is the top-level package of the ecosystem. It ties together the ice flow solver (`Huginn`), the mass balance models (`Muninn`), and the core data structures (`Sleipnir`) into an end-to-end differentiable glacier model. Its primary purpose is to enable **Universal Differential Equations (UDEs)**: hybrid models that combine physical PDEs with data-driven regressors such as neural networks, trained end-to-end with gradient-based optimization.

Reviewer: ```suggestion
[`ODINN.jl`](https://github.com/ODINN-SciML/ODINN.jl) is the top-level package of the ecosystem. It ties together the ice flow solver (`Huginn`), the mass balance models (`Muninn`), and the core data structures (`Sleipnir`) into an end-to-end differentiable glacier model. Its primary purpose is to enable inverse modelling via **Universal Differential Equations (UDEs)**: hybrid models that combine physical PDEs with data-driven regressors.
```

`ODINN` provides:

  - **Classical inversion** — fit scalar or spatially distributed parameters (e.g. the Glen flow rate factor `A` or sliding coefficient `C`) to observed ice thickness or surface velocity, without neural networks.
  - **Functional inversion (UDE training)** — replace or augment a physical law with a neural network, train it with automatic differentiation through the PDE solver.
  - A **law system** (`LawA`, `LawY`, `LawU`) that wraps neural networks or custom functions into the SIA2D solver in a composable, AD-compatible way.
  - A **loss system** (`LossH`, `LossV`, `LossHV`, `MultiLoss`) for fitting to thickness and/or velocity observations.

>   - **Sensitivity configuration** (`UDEparameters`, `sensealg`) exposing the full SciMLSensitivity adjoint zoo: continuous/discrete adjoints, Enzyme VJP, Mooncake, and manual adjoints.

Reviewer: ```suggestion
  - **Sensitivity configuration** (`UDEparameters`, `sensealg`) exposing the full differentiable programming ecosystem: continuous/discrete adjoints within SciMLSensitivity, automatic differentiation (e.g., Enzyme VJP, Mooncake), and manual adjoints.
```

> 

Reviewer 2: ```suggestion
These ingredients can be mixed together, for example `ODINN` can perform a functional inversion and a classical inversion at the same time by fitting initial conditions and a parameterization of the iceflow equations.
```

## Use directly vs. using lower-level packages

Use `ODINN.jl` (i.e. `using ODINN`) when you need:

  - End-to-end UDE training or classical inversion.
  - Gradient-based optimization through the glacier PDE.

>   - Access to the full `Inversion` workflow.

Reviewer 2: Quite vague

> Use `Huginn` alone when you only need **forward simulation** (no gradients, no NN training). Use `Muninn` alone for mass balance computation. Use `Sleipnir` alone for data structure manipulation. This modular structure means each downstream use case only pays the compilation cost of what it needs.

Reviewer: ```suggestion
Use `Huginn` alone when you only need **forward modelling**, which does not any form of  automatic differentiation. Use `Muninn` alone for mass balance computation. Use `Sleipnir` alone for data structure manipulation. This modular structure means each downstream use case only pays the compilation cost of what it needs.
```
